Gd 108

Barmalej24 opened the pull request.

Barmalej24 committed:
Add Domain Entity

Barmalej24 committed:
fix

Barmalej24 committed:
fix access modifier

Barmalej24 committed:
#GD-73 state: To Verify work 2h 40m
Added Separate domain layer,
Domain entities have been added for type, template and messages,
Added entity interface,
Added repository interface,
Added services interface.

Barmalej24 committed:
#GD-73 state: To Verify work 100m
Added Exceptions for all domain class
Added additional property IsRemove
Added metods Update and Delete
Changed class constructors
Remove metods Create

Barmalej24 committed:
#GD-73 state: To Verify
Change Exceptions

Barmalej24 committed:
#GD-73 state: To Verify
Fix Exception
Add StringExceptions

Barmalej24 committed:
#GD-73 state: To Verify
Change StringExceptions

Barmalej24 committed:
#GD-108 state: To Verify work 90m
Editing XML documentation
Fixed models and interfaces

Barmalej24 committed:
#GD-110 state: To Verify work 120m
Added Tests for Domain Models and Exceptions
Fix Exceptions

Barmalej24 committed:
#GD-108 state: To Verify work 15m
Add Documentation for Exceptions

Barmalej24 committed:
#GD-110 state: To Verify work 45m
fix Tests for Domain Models and Exceptions

oleggolen commented on NotificationMicroservice.Domain/Models/Message.cs:
```diff
@@ -0,0 +1,90 @@
+        /// Длинна строки описывающий направление отправки
+        /// </summary>
+        public const int MAX_DIRECTION_LENG = 10;
+
+        private Guid _id;
+        private MessageType _messageType;
+        private string _messageText;
+        private string _direction;
+        private DateTime _createDate;
```
Можно не использовать эти приватные поля, так как смысла они особого теперь не имеют

oleggolen commented on NotificationMicroservice.Domain/Models/Message.cs:
```diff
@@ -0,0 +1,90 @@
+            if (direction.Length > MAX_DIRECTION_LENG)
+            {
+                throw new MessageDirectionLengthException(ExceptionStrings.ERROR_DIRECTION_LENG, direction.Length.ToString());
+            }
```
Лучше здесь ещё добавить проверку на просто пробел и добавить это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageTemplate.cs:
```diff
@@ -0,0 +1,190 @@
+        /// <exception cref="MessageTemplateLanguageLengthException"></exception>
+        /// <exception cref="MessageTemplateNullOrEmptyException"></exception>
+        /// <exception cref="MessageTemplateUserNameNullOrEmptyException"></exception>
+        public MessageTemplate(Guid id, MessageType messageType, string language, string template, bool isRemove, string createUserName, DateTime createDate, string? modifyUserName, DateTime? modifyDate)
```
Добавить проверки на строки, содержащие просто пробел и внести это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageTemplate.cs:
```diff
@@ -0,0 +1,190 @@
+        /// <exception cref="MessageTemplateUserNameNullOrEmptyException"></exception>
+        public void Update(MessageType messageType, string language, string template, bool isRemove, string modifyUserName, DateTime modifyDate)
+        {
+            if (string.IsNullOrEmpty(language))
```
Добавить проверки на строки, содержащие просто пробел и внести это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageType.cs:
```diff
@@ -0,0 +1,159 @@
+        /// <exception cref="MessageTypeUserNameNullOrEmptyException"></exception>
+        public void Delete(string modifyUserName, DateTime modifyDate)
+        {
+            if (string.IsNullOrEmpty(modifyUserName))
+            {
+                throw new MessageTypeUserNameNullOrEmptyException(ExceptionStrings.ERROR_USERNAME, modifyUserName);
+            }
```
Проверку на содержание просто пробела и внести это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageType.cs:
```diff
@@ -0,0 +1,159 @@
+        /// <exception cref="MessageTypeUserNameNullOrEmptyException"></exception>
+        public void Update(string name, bool isRemove, string modifyUserName, DateTime modifyDate)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new MessageTypeNameNullOrEmptyException(ExceptionStrings.ERROR_TYPE_NAME, name);
+            }
+
+            if (name.Length > MAX_NAME_LENG)
+            {
+                throw new MessageTypeNameLengthException(ExceptionStrings.ERROR_TYPE_NAME_LENG, name.Length.ToString());
+            }
+
+            if (string.IsNullOrEmpty(modifyUserName))
+            {
+                throw new MessageTypeUserNameNullOrEmptyException(ExceptionStrings.ERROR_USERNAME, modifyUserName);
+            }
```
Добавить проверку на содержание просто пробелы и внести это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageType.cs:
```diff
@@ -0,0 +1,159 @@
+        /// <exception cref="MessageTypeUserNameNullOrEmptyException"></exception>
+        public MessageType(Guid id, string name, bool isRemove, string createUserName, DateTime createDate, string? modifyUserName, DateTime? modifyDate)
+        {
+            if (id == Guid.Empty)
```
Добавить проверку на содержание просто пробела и внести это в тесты, а также вынести дублирующийся код в приватный метод валидации или в валидатор

oleggolen commented on NotificationMicroservice.Domain/Models/MessageTemplate.cs:
```diff
@@ -0,0 +1,190 @@
+        public void Delete(string modifyUserName, DateTime modifyDate)
+        {
+
+            if (string.IsNullOrEmpty(modifyUserName))
+            {
+                throw new MessageTemplateUserNameNullOrEmptyException(ExceptionStrings.ERROR_USERNAME, modifyUserName);
+            }
```
Добавить проверку на пробел и внести это в тесты

oleggolen commented on NotificationMicroservice.Domain/Models/MessageTemplate.cs:
```diff
@@ -0,0 +1,190 @@
+                throw new MessageTemplateNullOrEmptyException(ExceptionStrings.ERROR_TEMPLATE, template);
+            }
+
+            if (string.IsNullOrEmpty(modifyUserName))
```
Вынести дублирующийся код в метод или в валидатор

oleggolen commented on NotificationMicroservice.Domain/Models/MessageTemplate.cs:
```diff
@@ -0,0 +1,190 @@
+using NotificationMicroservice.Domain.Exception.Resources;
+using NotificationMicroservice.Domain.Interfaces.Model;
+
+namespace NotificationMicroservice.Domain.Models
```
Пространство имёт должно быть Entities, так как в домене у нас сущности, а не модели, также следует переименовать папку

oleggolen commented on NotificationMicroservice.Domain/Models/Message.cs:
```diff
@@ -0,0 +1,90 @@
+using NotificationMicroservice.Domain.Exception.Resources;
+using NotificationMicroservice.Domain.Interfaces.Model;
+
+namespace NotificationMicroservice.Domain.Models
```
Пространство имёт должно быть Entities, так как в домене у нас сущности, а не модели, также следует переименовать папку

oleggolen commented on NotificationMicroservice.Domain/Models/MessageType.cs:
```diff
@@ -0,0 +1,159 @@
+    /// <summary>
+    /// Тип шаблона сообщения
+    /// </summary>
+    public class MessageType : IModifyEntity<string, Guid>
```
Пространство имёт должно быть Entities, так как в домене у нас сущности, а не модели, также следует переименовать папку

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Repository/IBaseRepository.cs:
```diff
@@ -0,0 +1,27 @@
+﻿namespace NotificationMicroservice.Domain.Interfaces.Repository
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Repository/IMessageRepository.cs:
```diff
@@ -0,0 +1,18 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Repository
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Repository/IMessageTemplateRepository.cs:
```diff
@@ -0,0 +1,34 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Repository
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Repository/IMessageTypeRepository.cs:
```diff
@@ -0,0 +1,34 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Repository
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Repository/IBaseRepository.cs:
```diff
@@ -0,0 +1,27 @@
+    /// </summary>
+    /// <typeparam name="T"> Тип Entity для репозитория. </typeparam>
+    /// <typeparam name="TKey"> Тип первичного ключа. </typeparam>
+    public interface IBaseRepository<T, TKey>
```
Убрать прификс Base, так как Base - это обычно суффикс для базовых абстрактных классов, а не интерфейсов

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Model/IEntity.cs:
```diff
@@ -0,0 +1,19 @@
+﻿namespace NotificationMicroservice.Domain.Interfaces.Model
```
Переименовать в Entities.Base

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Model/IEntity.cs:
```diff
@@ -0,0 +1,19 @@
+    /// Описание базовой Entity 
+    /// </summary>
+    /// <typeparam name="TKey"> Тип первичного ключа. </typeparam>
+    public interface IEntity<Tkey>
```
Добавить ограничение на Generic тип, чтобы были возможно только примитивные типы для корректной работы базы данных

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Model/IModifyEntity.cs:
```diff
@@ -0,0 +1,25 @@
+﻿namespace NotificationMicroservice.Domain.Interfaces.Model
```
Перенести в Entities.Base, а также вынести часть Entitties в отдельный проект доменного блока, так как в домене у нас сущности, а не модели

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Model/IModifyEntity.cs:
```diff
@@ -0,0 +1,25 @@
+    /// </summary>
+    /// <typeparam name="T"> Тип полей пользователей. </typeparam>
+    /// <typeparam name="TKey"> Тип первичного ключа. </typeparam>
+    public interface IModifyEntity<T, TKey> : IEntity<TKey>
```
Сделать ограничение на generic тип, чтобы могли передаваться значения только примитивного типа для корректной работы базы данных

oleggolen commented on NotificationMicroservice.Domain/Exception/Resources/ExceptionStrings.cs:
```diff
@@ -0,0 +1,16 @@
+﻿namespace NotificationMicroservice.Domain.Exception.Resources
+{
+    public class ExceptionStrings
```
Сделать класс статическим

oleggolen commented on NotificationMicroservice.Domain/Exception/Resources/ExceptionStrings.cs:
```diff
@@ -0,0 +1,16 @@
+﻿namespace NotificationMicroservice.Domain.Exception.Resources
```
Лучше назвать namespace Helpers, а не Resources

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageService.cs:
```diff
@@ -0,0 +1,23 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Services
```
Убрать промежуточное звено Interfaces, и лучше сделать пространство имён NotificationMicroservice.Domain.Services.Abstractions и вынести это в отдельный проект

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageService.cs:
```diff
@@ -0,0 +1,23 @@
+    /// <summary>
+    /// Описание методов для сервиса Сообшений (чтение)
+    /// </summary>
+    public interface IMessageService
```
Лучше назвать IMessageDomainService

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTemplateService.cs:
```diff
@@ -0,0 +1,44 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Services
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTemplateService.cs:
```diff
@@ -0,0 +1,44 @@
+    /// <summary>
+    /// Описание методов для сервиса Шаблонов сообщений
+    /// </summary>
+    public interface IMessageTemplateService
```
Лучше назвать IMessageTemplateDomainService

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTemplateService.cs:
```diff
@@ -0,0 +1,44 @@
+
+        /// <summary>
+        /// Добавление Шаблона сообщений
+        /// </summary>
```
Везде добавить CansellationToken

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageService.cs:
```diff
@@ -0,0 +1,23 @@
+    /// </summary>
+    public interface IMessageService
+    {
+        /// <summary>
+        /// Получение всех сообщений
+        /// </summary>
+        /// <returns>Коллекция сущностей Сообщей</returns>
+        Task<IEnumerable<Message>> GetAllAsync();
+
+        /// <summary>
+        /// Получение сообщения по идентификатору
+        /// </summary>
+        /// <param name="id">Идентификатор сущности</param>
+        /// <returns>Сущность Сообщение</returns>
+        Task<Message>? GetByIdAsync(Guid id);
```
Везде добавить CansellationToken

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTypeService.cs:
```diff
@@ -0,0 +1,44 @@
+﻿using NotificationMicroservice.Domain.Models;
+
+namespace NotificationMicroservice.Domain.Interfaces.Services
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTypeService.cs:
```diff
@@ -0,0 +1,44 @@
+    /// <summary>
+    /// Описание основных методов для сервиса Типов сообщений
+    /// </summary>
+    public interface IMessageTypeService
```
Переименовать в IMessageTypeDomainService

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/IMessageTypeService.cs:
```diff
@@ -0,0 +1,44 @@
+    /// </summary>
+    public interface IMessageTypeService
+    {
+        /// <summary>
+        /// Получение всех сущностей Типа сообщений
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<MessageType>> GetAllAsync();
+
+        /// <summary>
+        /// Получение сущности Типа сообщений по идентификатору
+        /// </summary>
+        /// <param name="id">Идентификатор сущности</param>
+        /// <returns>Сущность</returns>
+        Task<MessageType>? GetByIdAsync(Guid id);
+
+        /// <summary>
+        /// Добавление Типа сообщений
+        /// </summary>
+        /// <param name="messageType">Сущность</param>
+        /// <returns>Идентификатор сущности</returns>
+        Task<Guid> AddAsync(MessageType messageType);
+
+        /// <summary>
+        /// Обновление сущности Типа сообщений
+        /// </summary>
+        /// <param name="messageType">Сущность</param>
+        /// <returns>Булевое значение</returns>
+        Task<bool> UpdateAsync(MessageType messageType);
+
+        /// <summary>
+        /// Мягкое удаление сущности Типа сообщений
+        /// </summary>
+        /// <param name="messageType">Сущность</param>
+        /// <returns>Булевое значение</returns>
+        Task<bool> DeleteAsync(MessageType messageType);
```
Везде добавить CansellationToken

oleggolen commented on NotificationMicroservice.Domain/Interfaces/Services/ISenderService.cs:
```diff
@@ -0,0 +1,9 @@
+﻿namespace NotificationMicroservice.Domain.Interfaces.Services
```
Добавить префикс abstractions и вынести в отдельный проект из домена, так как репозитории не особо имеют к нему отношения

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageDirectionLengthException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageDirectionLengthException(string? message, string? value)
+            : base(message, value)
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageDirectionNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        public MessageDirectionNullOrEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class MessageGuidEmptyException : ArgumentException
+    {
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageDirectionNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения направления отправки
+    /// </summary>
+    public class MessageDirectionNullOrEmptyException : ArgumentException
```
Лучше в качестве базового выбрать ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class MessageGuidEmptyException : ArgumentException
```
Лучше в качестве базового выбрать ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageTextNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения текста отправляемого сообщения
+    /// </summary>
+    public class MessageTextNullOrEmptyException : ArgumentException
```
Лучше в качестве базового выбрать  ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageTextNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        public MessageTextNullOrEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        public MessageTemplateGuidEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class MessageTemplateGuidEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateLanguageLengthException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public MessageTemplateLanguageLengthException()
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и значением параметра
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTemplateLanguageLengthException(string? message, string? value) 
+            : base(message, value)
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и исключением 
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="innerException">Внутренне исключение</param>
+        public MessageTemplateLanguageLengthException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateLanguageNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение ограничения длинны кодировки языка
+    /// </summary>
+    public class MessageTemplateLanguageNullOrEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateLanguageNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// </summary>
+    public class MessageTemplateLanguageNullOrEmptyException : ArgumentException
+    {
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public MessageTemplateLanguageNullOrEmptyException()
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и значением параметра
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTemplateLanguageNullOrEmptyException(string? message, string? value) 
+            : base(message, value)
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и исключением 
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="innerException">Внутренне исключение</param>
+        public MessageTemplateLanguageNullOrEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// </summary>
+        public MessageTemplateNullOrEmptyException()
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение опустого значения теста шаблона
+    /// </summary>
+    public class MessageTemplateNullOrEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateUserNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения пользоватля
+    /// </summary>
+    public class MessageTemplateUserNameNullOrEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateUserNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// </summary>
+        public MessageTemplateUserNameNullOrEmptyException()
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class MessageTypeGuidEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException. Также похожие исключения можно выделить в иерархию

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// </summary>
+    public class MessageTypeGuidEmptyException : ArgumentException
+    {
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public MessageTypeGuidEmptyException()
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и значением параметра
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTypeGuidEmptyException(string? message, string? value)
+            : base(message, value)
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и исключением 
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="innerException">Внутренне исключение</param>
+        public MessageTypeGuidEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeNameLengthException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTypeNameLengthException(string? message, string? value)
+            : base(message, value)
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
```
Убрать неиспользуемые конструкторы

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения названия
+    /// </summary>
+    public class MessageTypeNameNullOrEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeUserNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения пользоватля
+    /// </summary>
+    public class MessageTypeUserNameNullOrEmptyException : ArgumentException
```
Использовать в качестве базового ArgumentNullException

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeUserNameNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// </summary>
+    public class MessageTypeUserNameNullOrEmptyException : ArgumentException
+    {
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public MessageTypeUserNameNullOrEmptyException()
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и значением параметра
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTypeUserNameNullOrEmptyException(string? message, string? value) 
+            : base(message, value)
+        {
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и исключением 
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="innerException">Внутренне исключение</param>
+        public MessageTypeUserNameNullOrEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать неиспользуемые конструкторы

oleggolen reviewed (changes requested):
Подведу имтог. Код написан очень хорошо, грамотно, работает, запускается, доменный слой выполнен корректно. Очень круто теперь написаны тесты, в правильной структуре и всё проверяют.
Стоит исправить для более хорошей структуры кода и его понимания:
1. Убрать из классов исключений лишние конструкторы, также как для исключений похожего типа сделать иерархию исключений
2. Привести структуру проектов к тому виду, который я показал в репозитории, это упростит чистаемость и навигацию в проектах, а также уберёт путанницу в названиях
3. Более корректно переименовать классы и интерфейсы, пространства имён и разделить это всё по проектам
4. Отднотивный код валидации сгруппировать по методам или валидаторам
Очень надеюсь, что полученные комментарии помогут написать легко читаемый и понимаемый код, убрать дублирование кода, и сделать его более структурированным

Barmalej24 committed:
#GD-121 state: To Progress work 1h 40m
Removed unused constructors from exceptions
Changed inherited exceptions for null exceptions
Changed the entity structure
Made changes to namespice
Moved interfaces from the domain to a separate project

Barmalej24 committed:
fix

Barmalej24 committed:
deleted abstraction

Barmalej24 committed:
full fix domain layer
